src/Crypto/Lol/Cyclotomic/Tensor/CTensor.hs

```diff
@@ -319,13 +327,13 @@ instance (Storable a, Storable b, CRNS a, CRNS b, CRTrans a, CRTrans b)
     (CT' b) <- gaussWrapper f
     return $ CT' $ zip a b
 
-mulGPow' :: (TElt CT r, Fact m) => CT' m r -> CT' m r
+mulGPow' :: (TElt CT r, Fact m, Additive r) => CT' m r -> CT' m r
```

Comment:
Perhaps all the `TElt` constraints here should be last, not first (because they go without saying).  The algebraic constraints on `r` are most important.